pkg/updatestrategy/node_pool_manager.go

```diff
@@ -368,17 +380,57 @@ var evictPod = func(client kubernetes.Interface, logger *log.Entry, pod *v1.Pod)
 			Name:      pod.Name,
 			Namespace: pod.Namespace,
 		},
+		DeleteOptions: &metav1.DeleteOptions{
+			GracePeriodSeconds: pod.Spec.TerminationGracePeriodSeconds,
+		},
 	}
 
 	err := client.CoreV1().Pods(pod.Namespace).Evict(eviction)
 	if err != nil {
 		return err
 	}
+	localLogger.Info("Evicting pod")
+
+	// wait for the pod to be actually evicted and gone from the node.
+	// It has TerminationGracePeriodSeconds time to clean up.
+	start := time.Now().UTC()
+	err = waitForPodTermination(client, *pod)
+	if err != nil {
+		localLogger.Warnf("Pod not terminated within grace period: %s", err)
+	}
+
+	localLogger.Infof("Pod evicted. (Observed termination period: %s)", time.Now().UTC().Sub(start))
+	return nil
+}
+
+// waitForPodTermination waits for a pod to be terminated by looking up the pod
+// in the API server.
+// It waits for up to TerminationGracePeriodSeconds as specified on the pod +
+// an additional eviction head room.
+// This is to fully respect the termination expectations as described in:
+// https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods
+func waitForPodTermination(client kubernetes.Interface, pod v1.Pod) error {
+	if pod.Spec.TerminationGracePeriodSeconds != nil {
```

Comment:
nit: I would rather change the if to check for `== nil` and then do an early return instead of having the entire method wrapped inan `if` statement.

Reply:
Fixed!